plugins/outputs/zabbix/zabbix.go

```diff
@@ -0,0 +1,164 @@
+package zabbix
+
+import (
+	_ "embed"
+	"sort"
+	"time"
+
+	"github.com/datadope-io/go-zabbix"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+// Zabbix allows pushing metrics to Zabbix software
+type Zabbix struct {
+	Host                  string
+	Port                  int
+	AgentActive           bool `toml:"agent_active"`
+	Prefix                string
+	SkipMeasurementPrefix bool `toml:"skip_measurement_prefix"`
+
+	// Config for LLD and autoregister. Needed here for config parsing.
+	LLDPeriod              int `toml:"lld_period"`
+	LLDResetPeriod         int `toml:"lld_reset_period"`
+	Autoregister           string
+	AutoregisterSendPeriod config.Duration
+
+	Log telegraf.Logger `toml:"-"`
+
+	// writeCount count the number of pushes to known where to return LLD data
+	writeCount int
+
+	// lldHandler handles low level discovery data
+	lldHandler ZabbixLLD
+
+	// autoregisterHandler handles host autoregistration
+	autoregisterHandler ZabbixAutoregister
+}
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (z *Zabbix) SampleConfig() string {
+	return sampleConfig
+}
+
+// Connect does nothing, Write() would initiate connection in each call.
+// Checking if Zabbix server is alive in this step does not allow Telegraf
+// to start if there is a temporal connection problem with the server.
+// Initilize LLD and autoregister maps. Copy config values to them. Configure Logger.
+func (z *Zabbix) Connect() error {
+	z.lldHandler = ZabbixLLD{}
+	z.lldHandler.Log = z.Log
+	z.lldHandler.resetPeriod = z.LLDResetPeriod
+	z.lldHandler.cache = make(map[uint64]bool, 50)
+	z.lldHandler.receivedData = make(map[HostLLD]LLDTags)
+	z.lldHandler.previousReceivedData = make(map[HostLLD]LLDTags)
+
+	z.autoregisterHandler = ZabbixAutoregister{}
+	z.autoregisterHandler.Log = z.Log
+	z.autoregisterHandler.autoregister = z.Autoregister
+	z.autoregisterHandler.autoregisterSendPeriod = z.AutoregisterSendPeriod
+	z.autoregisterHandler.autoregisterLastSend = make(map[string]time.Time)
+
+	return nil
+}
+
+func (z *Zabbix) Close() error {
+	return nil
+}
+
+// Write sends metrics to Zabbix server
+func (z *Zabbix) Write(metrics []telegraf.Metric) error {
+	if len(metrics) == 0 {
+		return nil
+	}
+
+	var zbxMetrics []*zabbix.Metric
+
+	for _, metric := range metrics {
+		hostname, ok := metric.Tags()["host"]
+		if !ok {
+			z.Log.Warnf("Missing hostname in metric %v. Ignoring.", metric)
+			continue
+		}
```

Comment:
You cannot do this see my other comment about `omit_hostname`.

Reply:
Fixed in 9dcd24db7